odoo/addons/base/models/res_device.py

```diff
@@ -153,7 +153,30 @@ def _revoke(self):
 
     @api.model
     def _select(self):
-        return "SELECT DISTINCT ON (D.user_id, D.session_identifier, D.platform, D.browser) D.*"
+        return """
+            SELECT DISTINCT ON (D.user_id, D.session_identifier, D.platform, D.browser)
+            abs(hashtext(
+                D.session_identifier || '_' ||
+                COALESCE(D.platform, 'unknown') || '_' ||
+                COALESCE(D.browser, 'unknown')
+            )) AS id,
```

Comment:
`hashtext` being an PostgreSQL internal function, I would instead use `md5` to get the hash. The risk of collisions is negligible.
You should also put the `user_id` in the key.
```suggestion
            CONCAT(
              'x',
              substr(
                md5(CONCAT(D.user_id, '_', D.session_identifier, '_', COALESCE(D.platform, 'unknown'), '_', COALESCE(D.browser, 'unknown'))),
                1, 8
              )
            )::bit(32)::int AS id,
```